x-pack/legacy/plugins/uptime/public/components/functional/filter_group/parse_filter_map.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+interface FilterField {
+  name: string;
+  fieldName: string;
+}
+
+export const parseFiltersMap = (
+  filterMapString: string,
+  fields: FilterField[]
+): Record<string, any[]> => {
+  const filterSlices: Record<string, any[]> = {};
+  try {
+    const map = new Map<string, string[]>(JSON.parse(filterMapString));
+    fields.forEach(({ name, fieldName }) => {
+      filterSlices[name] = map.get(fieldName) ?? [];
+    });
+    return filterSlices;
+  } catch (e) {
```

Comment:
Agree 100% (I came to the same conclusion while reviewing your comment above) we can simply `throw e` and it will be a much better experience for everyone when people encounter problems. Resolved in https://github.com/elastic/kibana/commit/3b16fc95438c4c7cf92bcccfc4234bf5c156a943.

Reply:
I'd actually like to chat about this. Do you think it's more useful to the user (and us) to throw the original error (i.e. `Unexpected character s in JSON at position 0`), or to throw a custom error like `Error parsing filter map`? If we throw from here we will get a stack trace that leads right back to this file still. If we don't throw a custom error I feel we should just remove the `try...catch` altogether.

Reply:
After discussion we've decided we don't need the `try..catch` at all.